### PR TITLE
👌 IMPROVE: Write subset of trajectory frames to disk.

### PR DESCRIPTION
**Pull request by mjclarke94**
Given some analysis work flows may only look at every nth frame, or discard some initial portion of data, allowing a trajectory file to be written which only includes the relevant data can significantly increase the speed of post-processing.

This keeps the same default behaviour so shouldn't impact preexisting workflows, but allows an integer to be passed to look at every nth step, or some other iterable to be passed in for more granular control.

### CODE REVIEW

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
thanks @mjclarke94, can you fix the pre-commit.
If you haven't used it before, its very easy: https://pre-commit.com/,
simply `pip install pre-commit` then run `pre-commit run -all` and it will run all the formatting and style checks, then you can also set it up to run before you commit, with `pre-commit install`

**Comment by chrisjsewell** [comment]
the python 3.6 tests install is failing, but anyhow the latest aiida-core no longer supports python 3.6, so I'm just going to quickly look into updating the testing

**Comment by mjclarke94** [comment]
> the python 3.6 tests install is failing, but anyhow the latest aiida-core no longer supports python 3.6, so I'm just going to quickly look into updating the testing

That's a testing strategy I can get behind

**Comment by chrisjsewell** [comment]
ok lets give this a go

File: aiida_lammps/data/trajectory.py [hunk 1]
```diff
@@ -213,8 +219,12 @@ def get_step_structure(
             original_structure=original_structure,
         )
 
-    def write_as_lammps(self, handle):
-        """Write out the lammps trajectory to file."""
-        for string in self.iter_step_strings():
+    def write_as_lammps(self, handle, steps=None):
+        """Write out the lammps trajectory to file.
+
```
Comment (chrisjsewell) [inline aiida_lammps/data/trajectory.py:224]:
```suggestion

```

**Pull request merged**